x/authz/client/cli/tx.go

```diff
@@ -82,6 +83,25 @@ Examples:
 
 			var authorization authz.Authorization
 			switch args[1] {
+			case "count":
+				msgType, err := cmd.Flags().GetString(FlagMsgType)
+				if err != nil {
+					return err
+				}
+
+				allowedAuthorizations, err := cmd.Flags().GetInt32(FlagAllowedAuthorizations)
+				if err != nil {
+					return err
+				}
+
+				if allowedAuthorizations <= 0 {
+					return fmt.Errorf("allowed-authorizations should be greater than zero")
+				}
+
+				// Increment allowed authorizations by one to ensure the correct desired count.
+				// This is because when the 'Msg' is dispatched to authz's 'Keeper.DispatchAction()'
+				// it calls 'CountAuthorization.Accept()' which first decrements allowedAuthorizations than validates.
+				authorization = authz.NewCountAuthorization(msgType, allowedAuthorizations + 1)
```

Comment:
I don't think this `+1` is correct. The `SendAuthorization` doesn't do anything like that, and this should be pretty much the same, except decrementing by 1 instead of by the amount being sent.

Reply:
Fixed